scanner/families/malware/yara/yara.go

```diff
@@ -130,13 +127,17 @@ func (s *Scanner) Run(ctx context.Context, sourceType utils.SourceType, userInpu
 			}
 		}
 
-		// Execute the yara command
-		// nolint:gosec
-		yaraCommand := exec.Command(yaraBinaryPath, args...)
-		err = utils.RunCommandAndParseOutputLineByLine(yaraCommand, parserFunc, errCheckFunc)
-		if err != nil {
-			s.sendResults(retResults, fmt.Errorf("failed to run yara command: %w", err))
-			return
+		// Define the yara args to run
+		args := []string{"-C", s.compiledRuleFile, "-r", "-w", "-m", "-p", strconv.Itoa(runtime.NumCPU())}
+		s.logger.Infof("Running yara...")
+
+		for _, d := range s.config.DirectoriesToScan {
+			yaraCommand := exec.CommandContext(ctx, yaraBinaryPath, append(args, fsPath+d)...)
```

Comment:
not really sure why we have the "directories to scan" at all since we can achieve the same behaviour by specifying those dirs through existing scan inputs for malware family. whats the difference between defining things to scan now? could you provide example configs?

Reply:
The `Inputs` that you are referring to I guess - https://github.com/openclarity/vmclarity/blob/main/scanner/families/malware/config.go#L27 ? - is the mount point where the whole filesystem will be mounted on the scanner VM.

This `Inputs` field gets set when the disks get attached to the scanner VM as stated in the comment here: https://github.com/openclarity/vmclarity/blob/main/orchestrator/watcher/assetscan/families.go#L171

So the `DirectoriesToScan` field is needed to narrow down the whole mounted filesystem in the `Inputs` field.

After this PR gets merged you can configure the malware scanner family to only scan these directories when `POST`-ing a new scan config.

```
"scanFamiliesConfig": {
        ...
        },
        "malware": {
          "enabled": true,
          "yara_directories_to_scan": ["/home", "/opt", ...]
        },
        ...
      }
}
```

Reply:
thanks @adamtagscherer, makes sense!